src/environmentd/src/http.rs

```diff
@@ -726,16 +746,31 @@ async fn init_ws(
                 let creds = Credentials::Token { token };
                 (creds, options)
             }
+            WebSocketAuth::OptionsOnly { options: _ } => {
+                anyhow::bail!("Missing auth information!");
+            }
+        };
+        (auth(frontegg, creds).await?, options)
+    } else if let Some(user) = existing_user {
+        // If we're not using frontegg and we have an existing user provided by an
+        // upstream middleware, just use the options field from the auth message
+        match ws_auth {
+            WebSocketAuth::OptionsOnly { options }
+            | WebSocketAuth::Basic {
+                user: _,
+                password: _,
+                options,
+            }
+            | WebSocketAuth::Bearer { token: _, options } => (user.clone(), options),
```

Comment:
The bearer case also seems like it should error because we never expect it.

Reply:
I added this in to avoid any frontend changes in the web console, since the frontend can still send a token along with it's initial auth message and we will just ignore it if the user is already set. That's what allowed me to be able to test this e2e without any console changes.

Happy to go either way on this -- if we assume the console will have a conditional around the auth logic when in user impersonation mode then being stricter here is fine